net/v2/convert.go

```diff
@@ -113,6 +114,40 @@ func IPFromAddr(addr netip.Addr) net.IP {
 	return net.IP(addr.AsSlice())
 }
 
+// IPFromInterfaceAddr can be used to extract the underlying IP address value (as a
+// net.IP) from the return values of net.InterfaceAddrs(), net.Interface.Addrs(), or
+// net.Interface.MulticastAddrs(). (net.Addr is also used in some other APIs, but this
+// function should not be used on net.Addrs that are not "interface addresses".)
+func IPFromInterfaceAddr(ifaddr net.Addr) net.IP {
```

Comment:
See https://github.com/danwinship/kubernetes/commit/922f93d5b21f7ced360c70553085f8801facb2e1

Reply:
seems very specific for our specific use case for a library, despite the name and the description clarifies that perfectly, the output of the InterfaceAddrs is `net.Addr` that is a very generic interface https://pkg.go.dev/net#Addr , it also may have protocol ip and port or ip and network